src/protocols/ToplevelMapping.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <vector>
+#include "WaylandProtocol.hpp"
+#include "hyprland-toplevel-mapping-v1.hpp"
+
+class CToplevelMappingManager {
+  public:
+    CToplevelMappingManager(SP<CHyprlandToplevelMappingManagerV1> resource_);
+
+    bool good() const;
+
+  private:
+    SP<CHyprlandToplevelMappingManagerV1> resource;
+};
+
+class CToplevelMappingProtocol : IWaylandProtocol {
+  public:
+    CToplevelMappingProtocol(const wl_interface* iface, const int& ver, const std::string& name);
+
+    virtual void bindManager(wl_client* client, void* data, uint32_t ver, uint32_t id);
+
+  private:
+    void                                                    onManagerResourceDestroy(CToplevelMappingManager* mgr);
+    void                                                    destroyHandle(CHyprlandToplevelWindowMappingHandleV1* handle);
+
+    std::vector<UP<CToplevelMappingManager>>                m_vManagers;
+    std::vector<SP<CHyprlandToplevelWindowMappingHandleV1>> m_vHandles;
```

Comment:
don't rawdog resource types, please. Wrap it like you did for the manager

Reply:
done